src/lambda/repo.js

```diff
@@ -0,0 +1,34 @@
+import fetch from 'node-fetch';
+
+exports.handler = async function(event, context, callback) {
+	const result = await repoInfo(
+		event.queryStringParameters.repo || 'preactjs/preact'
+	);
+	return {
+		statusCode: 200,
+		body: JSON.stringify(result)
+	};
+};
+
+function checkStatus(r) {
+	if (!r.ok) {
+		throw new Error(`${r.status}: Request failed for '${r.url}'`);
+	}
+	return r;
+}
+
+function memoize(fn) {
+	const CACHE = {};
+	return key => CACHE[key] || (CACHE[key] = fn(key));
+}
+
+// TODO: use GraphQL to avoid fetching so much data
+const repoInfo = memoize(repo =>
+	fetch(`https://api.github.com/repos/${repo}`)
+		.then(checkStatus)
+		.then(r => r.json())
+		.catch(() => ({
```

Comment:
Should we let this error bubble so we get telemetry if it starts failing and rely on the client to fallback to default values?

Reply:
I like that idea. IIRC the client code calling the GitHub API already has handling for non-ok status